PayPal TLS Fix

bencurthoys opened the pull request:
PayPal requires TLS 1.2 now - Added comments explaining this is the
case, and additional error handling for when the callback fails when TLS
1.0 or 1.1 is set.

Sorry for the whitespace changes; Visual Studio just reformatted the file. 

I added all the projects as references to the demo to make it easier to get started.

bencurthoys committed:
PayPal TLS Fix

PayPal requires TLS 1.2 now - Added comments explaining this is the
case, and additional error handling for when the callback fails when TLS
1.0 or 1.1 is set.

bencurthoys committed:
Removed Sandbox Credentials

Tch.

TerribleDev commented on src/Owin.Security.Providers.PayPal/PayPalAuthenticationHandler.cs:
```diff
@@ -81,9 +81,25 @@ public PayPalAuthenticationHandler(HttpClient httpClient, ILogger logger)
+                    {
+                        if (!System.Net.ServicePointManager.SecurityProtocol.HasFlag(System.Net.SecurityProtocolType.Tls12))
+                        {
+                            throw new Exception("PayPal requires TLS v1.2. TLS v1.0 and v1.1 connections will be refused. Set System.Net.ServicePointManager.SecurityProtocol = System.Net.ServicePointManager.SecurityProtocol | System.Net.SecurityProtocolType.Tls12", ex);
```
does this make more sense to be a WebException?

bencurthoys replied:
Yes. And I'll throw the original InnerException (which is a WebException) as the new InnerException, rather than having a WebException containing an HttpRequestException containing a different WebException.

TerribleDev commented on OwinOAuthProvidersDemo/App_Start/Startup.Auth.cs:
```diff
@@ -3,299 +3,305 @@
-      //});
-		}
-	}
+    public partial class Startup
```
what changed here? did you change the space delimiter by accident?

bencurthoys replied:
Certainly didn't change it on purpose. I will undo the whitespace changes once I figure out how to revert a file in Git. I'm used to Subversion.

TerribleDev replied:
I'm not super concerned if you can't revert the whitespace

bencurthoys committed:
Changed exception to WebException

If the PayPalAuthenticationHandler sees that an error caused by the
wrong SecurityProtocol has occurred, it throws a new WebException
instead of just Exception

bencurthoys committed:
Reverted Whitespace changes to Startup.Auth.cs

Whitespace somehow got changed. I think this reverts it.
